plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.test.ts

```diff
@@ -538,6 +559,59 @@ describe('NunjucksWorkflowRunner', () => {
       );
     });
 
+    it('should deal with checkpoints', async () => {
+      const task = {
+        ...createMockTaskWithSpec({
+          apiVersion: 'scaffolder.backstage.io/v1beta3',
+          parameters: {},
+          steps: [
+            {
+              id: 'test',
+              name: 'name',
+              action: 'checkpoints-action',
+              input: { foo: 1 },
+            },
+          ],
+          output: {
+            key1: '${{steps.test.output.key1}}',
+            key2: '${{steps.test.output.key2}}',
+            key3: '${{steps.test.output.key3}}',
+          },
+        }),
+        getTaskState: (): Promise<
+          | {
+              state: {
+                [key: string]:
+                  | { status: 'failed'; reason: string }
+                  | {
+                      status: 'success';
+                      value: JsonValue;
+                    };
+              };
+            }
+          | undefined
+        > => {
+          return Promise.resolve({
+            state: {
```

Comment:
Think this needs updating now right? Should be wrapped in `checkpoints` according to the latest changes in the BEP.